### PR TITLE
[silabs] reworked main.cpp 

### PR DESCRIPTION
**Pull request by Sarthak-Shaha**
Problem:
Silabs main.cpp not up to date with silabs template

Fix / solution:
Reworked silabs main.cpp to follow template. 
Moved functionality to matterconfig.cpp
Related modifications in gsdkspam.cpp and wiseMCUspam.cpp

### CODE REVIEW

**Review by jepenven-silabs — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: examples/platform/silabs/MatterConfig.cpp [hunk 1]
```diff
@@ -129,6 +155,42 @@ CHIP_ERROR SilabsMatterConfig::InitOpenThread(void)
 }
 #endif // CHIP_ENABLE_OPENTHREAD
 
+void application_start(void * unused)
```
Comment (selissia) [inline examples/platform/silabs/MatterConfig.cpp:158]: Should be in a namespace

**Review by selissia — CHANGES_REQUESTED** [review]
Needs a namespace statement

File: examples/platform/silabs/MatterConfig.cpp [hunk 2]
```diff
@@ -129,6 +155,44 @@ CHIP_ERROR SilabsMatterConfig::InitOpenThread(void)
 }
 #endif // CHIP_ENABLE_OPENTHREAD
 
+namespace {
+    void application_start(void * unused)
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/MatterConfig.cpp:159]: a namespace doesn't require ident for its content. Restyler is probably complaining about this

Comment (Sarthak-Shaha) [reply examples/platform/silabs/MatterConfig.cpp]: Yes. Updated.

**Review by selissia — APPROVED** [review]

**Pull request merged**